source/grid/tria.cc

```diff
@@ -10605,6 +10605,59 @@ Triangulation<dim,spacedim>::end_face () const
 }
 
 
+/*------------------------ Vertex iterator functions ------------------------*/
+
+
+template <int dim, int spacedim>
+typename Triangulation<dim,spacedim>::vertex_iterator
+Triangulation<dim,spacedim>::begin_vertex() const
+{
+  if (dim==1)
+    {
+      Assert(false, ExcNotImplemented());
+      return raw_vertex_iterator();
+    }
+  else
+    {
+      vertex_iterator i = raw_vertex_iterator(const_cast<Triangulation<dim, spacedim>*>(this),
+                                              0,
+                                              0);
+      if (i.state() != IteratorState::valid)
+        return i;
+      while (i->used() == false)
+        if ((++i).state() != IteratorState::valid)
+          return i;
```

Comment:
I think this loop is actually correct, because there presumably is always a used vertex. But it would be nice to at least have a comment to the effect that the loop will eventually terminate.